### PR TITLE
Uniswap Oracle

### PR DESCRIPTION
**Pull request by NickCuso**
Downsides of this approach:
 - Gas costs: approx 90k gas for this call. We may be able to reduce this a bit
 - The exchange rate will return as 0 for 24 hours after the first update. When integrating this, since for Unlock we are whitelisting supported tokens we could call update in the whitelist function itself. This would start the clock then instead of at the first token purchase. We could alternatively add logic to return the rate at the moment during that initial window.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     docker:
       - image: circleci/node:10
       - image: trufflesuite/ganache-cli:v6.9.1
-        command: ganache-cli -k istanbul
+        command: ganache-cli -k istanbul -e 10000000000
```
Comment (julien51) [inline .circleci/config.yml:20]: Is that change intentional?

File: src/protocols/uniswapOracle.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  /**
+   * Deploys a Uniswap oracle.
+   * @param uniswapFactory the uniswap factory address
+   * @returns the Uniswap oracle as a truffleContract object
+   */
+  deploy: async (web3, owner, uniswapFactory) => {
+    const oracle = await truffleContract.new(
+      web3,
+      uniswapOracleJson.abi,
+      `0x${uniswapOracleJson.bytecode.replace(/0x/, "")}`,
```
Comment (julien51) [inline src/protocols/uniswapOracle.js:14]: Does `bytecode` include multiple instances of `0x` ? If not, then you can just use `uniswapOracleJson.bytecode`

File: .circleci/config.yml (hunk 1 above)
Comment (NickCuso) [reply .circleci/config.yml:20]: Yea default balance is 100 eth. This may be overkill but allows the tests to fund more for a pair, given a more realistic exchange rate & slippage.

File: src/protocols/uniswapOracle.js (hunk 2 above)
Comment (NickCuso) [reply src/protocols/uniswapOracle.js:14]: That's fair, will change. It was copy pasted code to work with or without the 0x prefix. We don't need that here.
